scripts/shtf.py

```diff
@@ -0,0 +1,629 @@
+
+        body = {
+            "username": "beamer-monitoring-bot",
+            "channel": self.config["rocketchat"]["channel"],
+            "icon": ":heart",
+            "text": text,
+        }
+
+        r = requests.post(self.config["rocketchat"]["url"], data=json.dumps(body), headers=headers)
+
+        if r.status_code == 200:
+            return True
+
+        return False
+
+    def send_to_telegram(self, text: str) -> bool:
+        global notifications_sent
+
+        token = self.config["telegram"]["token"]
+        chat_id = self.config["telegram"]["chat_id"]
+
+        url = f"https://api.telegram.org/bot{token}/sendMessage?chat_id={chat_id}&text={text}"
+        r = requests.get(url).json()
+
+        if r["ok"]:
+            return True
+
+        return False
+
+
+notifications_sent = 0
+timeout_between_sends = 61
+claimRequestExtension = 86400
+global_config: Optional[ConfigType] = None
+
+
+def _default_config() -> dict:
+    return {"chains": {}, "tokens": {}}
+
+
+def set_config(path: Path) -> None:
+    global global_config
+
+    config = _default_config()
+
+    if path is not None:
+        config = _merge_dicts(config, toml.load(path))
+
+    rpcs = {}
+    explorers = {}
+    for _key, chains in config["chains"].items():
+        chain_id = int(chains["chain-id"])
+
+        rpcs[chain_id] = (
+            chains["rpc-url"],
+            BlockNumber(int(chains["start-block"])),
+        )
+        explorers[chain_id] = chains["explorer"]
+
+    global_config = {
+        "agent_address": config["agent-address"].lower(),
+        "deployment_dir": Path(config["deployment-dir"]),
+        "notification_system": config["notification-system"],
+        "rpcs": rpcs,
+        "explorers": explorers,
+        "notification": config["notification"],
+        "tokens": config["tokens"],
+    }
+
+
+def get_config() -> ConfigType:
+    assert global_config is not None
+    return global_config
+
+
+@click.command
+@click.option(
+    "--config",
+    "-c",
+    "config_path",
+    type=click.Path(exists=True, file_okay=True, dir_okay=False, path_type=Path),
+    metavar="PATH",
+    help="Config file with chains configuration",
+)
+def main(
+    config_path: Path,
+) -> None:
+    setup_logging(log_level="DEBUG", log_json=False)
+
+    set_config(config_path)
+
+    transfers = create_transfers_object(fetch_events())
+
+    cleanup_transfers(transfers)
+
+    ctx = Context()
+    ctx.agent_address = get_config()["agent_address"]
+
+    analyze_transfers(transfers, ctx)
+
+
+def cleanup_transfers(transfers: transfersType) -> None:
+    for requestId, values in transfers.copy().items():
+        if values.get("created") is None:
+            del transfers[requestId]
+
+        claims = values.get("claimed")
+        if claims is not None:
+            for events in claims.values():
+                events.sort(key=lambda a: -1 if a.last_challenger == ADDRESS_ZERO else 1)
+
+
+def fetch_events() -> ChainToEvent:
```

Comment:
you can give the config to this as argument so we dont execute get_config that much

Reply:
Fetch events is executed just once, so there is no real difference if we call get_config in it or outside of it. I could do
`config = get_config()` at the beginning of the function and it will have the same effect.

Passing the config as an argument to the function only makes it easier to test, as you could pass a different test config. (and not have to  monkeypatch the get_config)

Reply:
I believe you can define config at start at least. We dont want to read from file everytime. Maybe a small performance increase but still.

Reply:
It's not read from file. set_config reads the file and sets it in the global_conf. From there on it's read from memory.